[Cmake] Bump dav1d 1.5.2

fuzzard opened the pull request:
## Description
Bump dav1d 1.5.2 for all platforms

## Motivation and context
Test further meson build infrastructure. Helps to refine the infrastructure  in #27369 to another library.
Updated dav1d

## How has this been tested?
Windows/macos

## What is the effect on users?
N/A

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[cmake] Introduce FindMeson package

Commit:
[cmake] Introduce FindNinja module

Commit:
[win] packagedownload extract msi

Add ability to extract an msi, and partial re-arrange folder for
meson msi explcitly. The msi structure meson comes with isnt "standard" to other
msi's inspected, so explicitly mention this use case

Commit:
[win] build tool - add meson 1.9.1-x64

Commit:
[tools/depends][native] Add pkgconf package

Primary intent is for Windows to build, but can be manually built for
unix platforms

Commit:
[cmake] Add FindPkgConfig module

native cmake FindPkgConfig does not recognise the pkgconf executable, so
we look to override to handle both pkgconfig and pkgconf

Commit:
[win] get_formed add alternative msi extraction tool (lessmsi)

add an option to use https://github.com/activescott/lessmsi to extract an msi instead of
msiexec.
Permissions and how CI jobs are run dont allow msiexec to succeed. (Service accounts,
TRUSTED_INSTALLER perms among the reasons).
General recommendations for most would be just to use msiexec administrative install

Commit:
[cmake] FindDav1d use new default structure for building/searching packages

Commit:
[win] Builddependencies remove prebuilt dav1d

Commit:
[cmake] FindGASPP build tool module

Find Gas-preprocessor script

Commit:
[dependencies] Bump libdav1d 1.5.2

Commit:
[cmake] FindFFMPEG add dependency tracking for dav1d dependency

Respect dependency force rebuild

fuzzard commented on cmake/modules/buildtools/FindPkgConfig.cmake:
```diff
@@ -0,0 +1,83 @@
+    execute_process(
+      COMMAND ${CMAKE_COMMAND} --build .
+      WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/pkgconf
+    )
```
actually a valid suggestion, but the implemntation is wrong

Commit:
[win] build tool - add nasm 3.0.1

Commit:
[cmake] FindNASM add native tools HINTS path

thexai commented:
> No idea what's really causing the uwp failure to crop their head up now

It's not just in UWP, nasm isn't used in regular x64 either.

fuzzard commented:
Uwp issue is a linking issue. Was missing some alloc/free symbols to build ffmpeg with open SSL, added the onecore patch, now it's failing with CMT. My guess is dav1d/ffmpeg aren't building as uwp, but don't know which, or why, yet. Will dig into it later in the week

thexai commented on project/BuildDependencies/scripts/0_package.native-win32.list:
```diff
@@ -6,5 +6,9 @@
+; lessmsi is a prerequisite to extract any msi packages for CI. Keep first in list
+lessmsi-v2.10.3.zip https://github.com/activescott/lessmsi/releases/download/v2.10.3
+meson-1.9.1-64.msi https://github.com/mesonbuild/meson/releases/download/1.9.1
+nasm-3.01-win32.zip https://www.nasm.us/pub/nasm/releasebuilds/3.01/win32
```
From some time is a build requirement use Windows x64 only. 

Changed here:
https://github.com/xbmc/xbmc/commit/52638ace6b63aa9d34e75d06c213a11b9a9affda

Then all build tools can be x64 (even if win32 also works). For me is fine just change link here.

Commit:
[cmake] FindVcvars introduce module

Upstream source is https://github.com/scikit-build/cmake-FindVcvars
BSD-3 licensed. We use as is. There was an effort for the creators to upstream into
cmake itself, but so far hasnt materialised. For now, add in tree.

thexai reviewed (approved):
Tested on Xbox Series S and confirmed is working (playback AV1 video).

Tested also in debug mode in Windows 11 PC with Visual Studio. Tested AV1 playback with DXVA2 enabled and disabled and confirmed HW acceleration is working, also SW decoding using ASM optimizations (low CPU usage even playback 4K videos). 

Good job!

Commit:
[cmake] ModuleHelper function to generate dev_env variable for externalproject_add

Currently specifically for windows to set a dev environment for target configure and
build steps for externalproject_add calls

Commit:
[CI] Test different ffmpeg build options

linux/webos - tools/depends/target testing the makefile setup - tests x64/arm/aarch64
freebsd - tests system ffmpeg
osx x64 - test explicit ENABLE_INTERNAL_FFMPEG
Other unix - tests search and fallback for KODI_DEPENDSBUILD auto build

Commit:
[cmake] meson cross generation - fix module specific flags when empty spaces

Remove extra whitespace from flag variables to avoid adding empty '' arguments

Commit:
[cmake] FindFFMPEG check VDPAU/VAAPI if optional to platform
